tools/osbuild-mpp

```diff
@@ -400,13 +400,18 @@ class ImageManifest:
         return ImageManifest._arch_from_rpm.get(rpm_arch, rpm_arch)
 
     @staticmethod
-    def load(imagename, tag=None, digest=None):
+    def load(imagename, tag=None, digest=None, transport=None):
+        if transport == "containers-storage":
+            protocol = "containers-storage:"
+        else:
+            protocol = "docker://"
```

Comment:
We should have a set of allowed values and error out if it's invalid, instead of falling back to docker for anything other than `containers-storage`.  The way it is now seems very error prone.  A minor typo can cause a lot of headaches, especially if the ref is valid for both transports.  Imagine wanting to build an image with a local (perhaps older or modified) version of a container image with ref `registry.example.com/example/image` and you accidentally type `"transport": "contianers-storage"`.  The build will pull `docker://registry.example.com/example/image` and successfully build the image, but wont be immediately clear why the embedded container is different from the one in the local storage.

Reply:
I could raise a RuntimeError if it works for you. What do you mean with embedded container?

Reply:
> I could raise a RuntimeError if it works for you.

That works.

> What do you mean with embedded container?

Container resolution in osbuild was originally used to embed containers in virtual disk images, so you could, for example, build a Fedora qcow2 and have containers in the base image, ready to run.  I still think of the container resolver being used for that purpose by default so I made up my example based on that.

Reply:
The use case it is the same but for demo purposes we would like to be able to build our container locally and then building our image with the container inside instead of building the container, pushing to a registry and downloading it.

Reply:
I've add a runtime error for the values that are not "docker" or "container-storage"

Reply:
Thank you!